### PR TITLE
Feature/253 edit notes

### PR DESCRIPTION
**Pull request by mkowalczyk97**
## Changes made

- I've added icon for editing note
- I've created screen view for editing note (title and discription for editing notes from categories: 'Without Category", "Diaries", "To Do" and just description for "To Pack" category)
- I've created editing notes logic in actions and reducer
- I've fixed bug where you couldn't delete notes from three categories (expect To-Pack)
- I've fixed bug with wrong import of navigationButton (this bug appeard in develop branch)
- I've  created correct validation of eddited note
- I've used Prettier and ESLint

**THERE IS BUG that i couldn't fixed - In order to see edited note you have to open 'Notes' again. Propably some problem with states (old issue)**

## Screenshots

(prefer animated gif)

## Test path (if necessary)

1. Go to Notes screen
2. Click on edit button
3. Edit your note
4. Go out of notes screen
5. Open notes screen again in order to see changes
6. Try this for notes from every category
...

## Checklist

- [ ] merged develop branch into feature branch
- [x] tested locally
  - [ ] in emulator
  - [x] on a real device
- [ ] added new dependencies (what deps?)
- [x] used ESLint (linted) and Prettier (formatted)
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Review by CalderBriar — APPROVED** [review]
Solid work. It must have taken time to go through the files to use prettier on them.

File: src/domains/notes/containers/EditNoteContainer.js [hunk 1]
```diff
@@ -17,11 +17,12 @@ import {
 
   const EditNoteContainer = (props) => {
 
-
+    const { noteId } = props;
+    console.log(noteId)
```
Comment (weronikaolejniczak) [inline src/domains/notes/containers/EditNoteContainer.js]: console.log out

File: src/domains/notes/containers/EditNoteContainer.js [hunk 2]
```diff
@@ -37,9 +72,28 @@ import {
...
+          });
+        }
+        setIsLoading(false);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+      }, [titleIsValid,
+        descriptionIsValid,
+        tripId,
+        noteId,
+        title,
+        'To Do',
```
Comment (weronikaolejniczak) [inline src/domains/notes/containers/EditNoteContainer.js]: czemu string jako dependency?

File: src/reducers/commonReducer.js [hunk 3]
```diff
@@ -75,6 +75,13 @@ export default (state = initialState, action) => {
         );
         break;
 
+      case EDIT_NOTE:
+        draft.trips[tripIndex].notes = [
```
Comment (weronikaolejniczak) [inline src/reducers/commonReducer.js]: tego raczej nie potrzeba, bo immer sam nam kopiuje stan

File: src/actions/notesActions.js [hunk 4]
```diff
@@ -129,27 +129,24 @@ export const editNoteRequest = (tripId, noteId, title, category, description) =>
...
+      category,
+      title,
+      description,
+    }).then(() => {
+      const newNote = new Note(
+        noteId,
         date,
-        category,
         title,
+        'To Pack',
```
Comment (weronikaolejniczak) [inline src/actions/notesActions.js]: znowu, czemu string w dependency array?

**Review by weronikaolejniczak — APPROVED** [review]

**Comment by weronikaolejniczak** [comment]
> Solid work. It must have taken time to go through the files to use prettier on them.

That's automatized.

**Pull request merged**